components/team_sidebar/selectors.js

```diff
@@ -0,0 +1,23 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+import {getJoinableTeams, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+
+export const moreTeamsToJoin = createSelector(
```

Comment:
Sorry for late review.
Looks like we could use this selector to SidebarRightMenu and SidebarHeaderDropdown as well.